spring-kafka/src/main/java/org/springframework/kafka/core/CompositeKafkaStreamsCustomizer.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.core;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.kafka.streams.KafkaStreams;
+
+/**
+ * Composite {@code KafkaStreamsCustomizer} customizes {@link KafkaStreams} by delegating.
+ *
+ * @author Nurettin Yilmaz
```

Comment:
`@since 2.1.5`

Also sounds like a wrong package for this API.
Would be better to have it in the `config`, but then we will have a package tangle with the `core` via `StreamsBuilderFactoryBean`.

I think we don't have choice now, but we can reconsider the place for these Kafka Streams related classes in the next `2.2` version.